Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -29,7 +65,31 @@ public void AddItem(Item item)
         /// </summary>
         public void RemoveItem(Item item)
         {
+            _items = RemoveFromArray(item);
+        }
+
+        private Item[] RemoveFromArray(Item item)
+        {
+            if (_items.Length == 0)
+            {
+                return _items;
+            }
+
+            var newItemArray = new Item[_items.Length - 1];
+            var indexItem = Array.IndexOf(_items, item);
+            for (int i = 0; i < _items.Length - 1; i++)
+            {
+                if (i < indexItem)
+                {
+                    newItemArray[i] = _items[i];
+                }
+                else if (i > indexItem)
+                {
+                    newItemArray[i - 1] = _items[i];
+                }
+            }
```

Comment:
Using Array.IndexOf() here is waste of performance. You can do that in your own for loop.

Reply:
solved